File: dev/20190624/k2works/src/statement.js
import createStaementData from "./createStatementData";

export const plays = {
  hamlet: { name: "Hamlet", type: "tragedy" },
  "as-like": { name: "As You Like It", type: "comedy" },
  othello: { name: "Othello", type: "tragedy" }
};
export const invoices = [
  {
    customer: "BigCo",
    performances: [
      {
        playID: "hamlet",
        audience: 55
      },
      {
        playID: "as-like",
        audience: 35
      },
      {
        playID: "othello",
        audience: 40
      }
    ]
  }
];

export function statement(invoice, plays) {
  return renderPlainText(createStaementData(invoice, plays));
}

function renderPlainText(data, invoice, plays) {
  let result = `Statement for ${data.customer}\n`;
  for (let perf of data.performances) {
    // print line for this order
    result += ` ${perf.play.name}: ${usd(perf.amount)}(${
      perf.audience
    } seats)\n`;
  }
  result += `Amount owed is ${usd(data.totalAmount)}\n`;
  result += `You earned ${data.totalVolumeCredits} credits\n`;
  return result;

  function usd(aNumber) {
    return new Intl.NumberFormat("en-US", {
      style: "currency",
      currency: "USD",
      minimumFractionDigits: 2
    }).format(aNumber / 100);
  }
}

File: dev/20190701/k2works/api/rental/src/main/java/rental/Statement.java
package rental;

public class Statement {
  Rentals rentals;

  Statement(Rentals rentals) {
    this.rentals = rentals;
  }

  String statement() {
    String result = rentals.customer.name() + "様のレンタル明細\n";

    for (Rental each : rentals.asList()) {
      result += "\t" + each.movie().title() + "\t" + each.amount() + "円\n";
    }

    int totalAmount = rentals.totalAmount();
    int points = rentals.totalPoints();

    result += "合計金額 " + totalAmount + "円\n";
    result += "獲得ポイント " + points + "p";
    return result;
  }
}
